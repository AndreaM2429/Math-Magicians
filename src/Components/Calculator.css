.result {
  background-color: grey;
  height: 15vh;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding-right: 10px;
}

span {
  font-size: 35px;
  color: #ffff;
}
