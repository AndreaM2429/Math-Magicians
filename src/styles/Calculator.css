.calculatorPage {
  flex-direction: column;
  width: 85%;
  padding: 15px;
}

.calculator {
  border: 8px solid gray;
  border-radius: 8px;
}

.result {
  background-color: grey;
  height: 50px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding-right: 10px;
}

span {
  font-size: 35px;
  color: #ffff;
}

@media (min-width: 768px) {
  .calculatorPage {
    flex-direction: row;
    justify-content: space-around;
    align-items: start;
  }

  .calculatorPage h3 {
    font-size: 30px;
  }
}
