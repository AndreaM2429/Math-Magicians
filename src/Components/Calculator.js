import calculate from '../Logic/Calculate';
import './Calculator.css';
import Panel from './Panel';

function Calculator() {
  return (
    <section>
      <div className="result">
        <span>0</span>
      </div>
      <Panel buttonEvents={calculate} />
    </section>
  );
}

export default Calculator;
