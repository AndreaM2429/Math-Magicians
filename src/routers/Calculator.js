import Calculator from '../Components/Calculator';

function calcularor() {
  return (
    <div className="calculatorPage d-flex">
      <h3>Let&apos;s do some math!</h3>
      <Calculator />
    </div>
  );
}

export default calcularor;
