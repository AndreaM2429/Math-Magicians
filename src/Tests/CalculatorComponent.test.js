// import react-testing methods
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
// userEvent library simulates user interactions.
import userEvent from '@testing-library/user-event';
// add custom jest matchers from jest-dom
import '@testing-library/jest-dom';
// the component to test
import Calculator from '../routers/Calculator';

describe('Test components', () => {
  test('Simulate user iteractions case 1', () => {
    // Render a React element into the DOM
    render(<Calculator />);

    userEvent.click(screen.getByText('7'));
    userEvent.click(screen.getByText('+'));
    userEvent.click(screen.getByText('8'));
    userEvent.click(screen.getByText('='));
    userEvent.click(screen.getByText('x'));
    userEvent.click(screen.getByText('5'));
    userEvent.click(screen.getByText('='));
    userEvent.click(screen.getByText('='));

    expect(screen.getByTitle('span')).toHaveTextContent('0');
  });

  test('Simulate user iteractions case 2', () => {
    // Render a React element into the DOM
    render(<Calculator />);

    userEvent.click(screen.getByText('7'));
    userEvent.click(screen.getByText('+'));
    userEvent.click(screen.getByText('8'));
    userEvent.click(screen.getByText('='));

    // assert that the alert message is correct using
    // toHaveTextContent, a custom matcher from jest-dom.
    expect(screen.getByTitle('span')).toHaveTextContent('15');
  });

  test('Render correctly', () => {
    // Render a React element into the DOM
    const page = renderer
      .create(<Calculator />)
      .toJSON();

    expect(page).toMatchSnapshot();
  });
});
