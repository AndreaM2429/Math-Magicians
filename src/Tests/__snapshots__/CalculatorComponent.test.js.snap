// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test components Render correctly 1`] = `
<div
  className="calculatorPage d-flex"
>
  <h3>
    Let's do some math!
  </h3>
  <section
    className="calculator"
  >
    <div
      className="result"
    >
      <span
        title="span"
      >
        0
      </span>
    </div>
    <div
      className="panel"
    >
      <button
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="operators"
        name="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="operators"
        name="x"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="operators"
        name="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="operators"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="special"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="operators"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </section>
</div>
`;
